scanners/nessus/nessus_none.py

```diff
@@ -93,6 +102,25 @@ def scan_id(self) -> int:
             raise RuntimeError("scan_id is None")
         return self._scan_id
 
+    @generic_authentication_factory()
+    def authentication_factory(self):
+        """This is the default function, attached to error reporting"""
+        raise RuntimeError(f"No valid authenticator found for {self}")
+
+    @authentication_factory.register(None)
+    def authentication_set_anonymous(self):
+        """No authentication: don't do anything"""
+        logging.info("NOT configured with any authentication")
+        return False
+
+    @authentication_factory.register("http_header")
+    def authentication_set_http_header_auth(self):
+        """
+        Nessus Professional doesn't support HTTP header (e.g. Authorization: Bearer token) auth
+        """
+
+        raise RuntimeError("http_header authentication is not supported in Nessus Professional.")
```

Comment:
I don't think rapidast knows necessarily that the nessus instance it's connecting to is Professional or not. I would simplify this so that it tells the user what combination of config options they have used is not supported. If that's done in above in the `authentication_factory()` method then this method is probably not needed.

Reply:
good idea. will update in this way